lib/interface_quda.cpp

```diff
@@ -108,6 +108,30 @@ CloverField *cloverEigensolver = nullptr;
 GaugeField momResident;
 GaugeField *extendedGaugeResident = nullptr;
 
+/** 
+ * callMultiSrcQuda related gauge split
+ * update_split_gauge : 
+ *   QUDA_UPDATE_SPLITE_GAUGE_OFF delete buffer after usage
+ *   QUDA_UPDATE_SPLITE_GAUGE_FALSE split gauge in buf
+ *   QUDA_UPDATE_SPLITE_GAUGE_TRUE  split gauge not in buf, need to call split
+ * update_split_gauge set to QUDA_UPDATE_SPLITE_GAUGE_OFF if QudaGaugeParam.use_split_gauge_bkup == QUDA_BOOLEAN_FALSE, QUDA_UPDATE_SPLITE_GAUGE_TRUE if QudaGaugeParam.use_split_gauge_bkup == QUDA_BOOLEAN_TRUE
+ * update_split_gauge != QUDA_UPDATE_SPLITE_GAUGE_OFF and split gauge in buf, the original links will swap with *_bkup 
+ * split_grid_bkup will be used to check whether split layout is changed or not
+ * change in gauge precsions will need loadGaugeQuda which results in re-distribute
+ * assumed no change in clover parameters
```

Comment:
I modified the comment here to make it clearer - but please correct me if I say anything wrong.
```suggestion
 * - QUDA_UPDATE_SPLITE_GAUGE_TRUE: the input gauge fields will be split and the buffered (split) gauges will be updated accordingly;
 * - QUDA_UPDATE_SPLITE_GAUGE_FALSE: the input gauge fields will not be split and the buffered (split) gauges will be used for split grid solves;
 * - QUDA_UPDATE_SPLITE_GAUGE_OFF: nothing will be done.
 
 * split_grid_bkup will be used to check whether split layout is changed or not
 * change in gauge precsions will need loadGaugeQuda which results in re-distribute
 * assumed no change in clover parameters
```